src/Orleans.Core/Messaging/MessageFactory.cs

```diff
@@ -20,11 +22,13 @@ public MessageFactory(SerializationManager serializationManager, ILogger<Message
 
         public Message CreateMessage(InvokeMethodRequest request, InvokeMethodOptions options)
         {
+           
             var message = new Message
             {
                 Category = Message.Categories.Application,
                 Direction = (options & InvokeMethodOptions.OneWay) != 0 ? Message.Directions.OneWay : Message.Directions.Request,
                 Id = CorrelationId.GetNext(),
+                ActivityId = GuidId.GetGuidId(EventSource.CurrentThreadActivityId),
```

Comment:
I think this should be a regular Guid. `GuidId` would introduce an allocation (it's essentially boxing a Guid) + indirection which I'm not sure is necessary.

Reply:
but using a Guid will make message header always has this field, not sure if we want to parse an empty ActivityId if users didn't set one.

Reply:
Oh right, I meant to not set it if the CurrentThreadActivityId is empty, hence the using GuidId instead of Guid, since some headers may not have this field. But failed to do it yet.